### PR TITLE
vector: 0.21.0 -> 0.21.1

### PR DESCRIPTION
**Pull request by happysalada**
###### Description of changes

This fixes a small regression. https://vector.dev/releases/

@kira-bruneau noticed you posted something last time about darwin breakage. I couldn't find a way to fix the breakage on darwin and was imagining that most usage was on linux anyway. Sorry about the disturbance!

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
@GrahamcOfBorg eval

**Comment by kira-bruneau** [comment]
> @kira-bruneau noticed you posted something last time about darwin breakage. I couldn't find a way to fix the breakage on darwin and was imagining that most usage was on linux anyway. Sorry about the disturbance!

Oh no problem! That's probably still a fair assumption 😅. We use vector as part of our dev flake at work and I just noticed the darwin build broke trying to do an update. It's not exactly a huge deal though, since we can just to pin to the old version for now.

Do you have an idea of what was wrong with using kafka? I can spend a bit of time looking into this at some point 🙂 I kind of let my maintenance slide though, and I have a blacklog of other things I probably want to get through first 😓

I'm just waiting on the Linux build to finish right now, but it looks like you just pushed some changes to disable the tests 😅

**Comment by happysalada** [comment]
right, this new version has 100 tests failing out of 1100. 
I'm not too happy about it, but I have to say that the tests were already failing on darwin for a long time.

on darwin the kafka failure was a weird symbol missing. It made me think it was a matter of the sdk used in nix being too old.
Removing the kafka feature another thing broke (If I remember correctly it was during zstd-sys build. Since the failure was not immediately obvious, I just gave up).

no rush on this. If you ever have time, great, however there might be more important stuff.
I'm there if you need.

**Comment by kira-bruneau** [comment]
Result of `nixpkgs-review pr 169934` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vector</li>
  </ul>
</details>

**Review by kira-bruneau — APPROVED** [review]
The Linux build looks ok with the tests disabled, but I'll let you decide when you want to merge it in case you're still trying to figure out the test failures.

**Comment by happysalada** [comment]
Thanks a lot for the review. It looks like it's working, so I'm going to merge this and revert If I see weird failures when using it on a server.

I'll most probably give the tests another go on the next version.

**Pull request merged**
